packages/web3-core/src/web3_subscriptions.ts

```diff
@@ -81,7 +82,10 @@ export abstract class Web3Subscription<
 			}
 		};
 
-		(this._requestManager.provider as Web3BaseProvider).on<Log>('message', messageListener);
+		(this._requestManager.provider as Web3BaseProvider).on<Log>(
+			'message',
+			messageListener as Web3ProviderEventCallback<Log>,
```

Comment:
I think the casting is not needed
```suggestion
		(this._requestManager.provider as Web3BaseProvider).on<Log>('message', messageListener);
```